go-controller/pkg/factory/factory.go

```diff
@@ -1,118 +1,93 @@
 package factory
 
 import (
+	"fmt"
 	"time"
 
-	utilwait "k8s.io/apimachinery/pkg/util/wait"
 	informerfactory "k8s.io/client-go/informers"
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/tools/cache"
+)
+
+// WatchFactory initializes and manages common kube watches
+type WatchFactory struct {
+	iFactory  informerfactory.SharedInformerFactory
+	informers map[string]cache.SharedIndexInformer
+}
 
-	"github.com/Sirupsen/logrus"
-	"github.com/openvswitch/ovn-kubernetes/go-controller/pkg/cluster"
-	"github.com/openvswitch/ovn-kubernetes/go-controller/pkg/kube"
-	"github.com/openvswitch/ovn-kubernetes/go-controller/pkg/ovn"
+const (
+	typePods       = "pods"
+	typeServices   = "services"
+	typeEndpoints  = "endpoints"
+	typePolicies   = "policies"
+	typeNamespaces = "namespaces"
+	typeNodes      = "nodes"
 )
 
-// Factory initializes and manages the kube watches that drive an ovn controller
-type Factory struct {
-	KClient        kubernetes.Interface
-	IFactory       informerfactory.SharedInformerFactory
-	ResyncInterval time.Duration
+// NewWatchFactory initializes a new watch factory
+func NewWatchFactory(c kubernetes.Interface, stopChan <-chan struct{}) (*WatchFactory, error) {
+	iFactory := informerfactory.NewSharedInformerFactory(c, 10*time.Minute)
+	wf := &WatchFactory{
+		iFactory:  iFactory,
+		informers: make(map[string]cache.SharedIndexInformer),
+	}
+
+	wf.informers[typePods] = iFactory.Core().V1().Pods().Informer()
+	wf.informers[typeServices] = iFactory.Core().V1().Services().Informer()
+	wf.informers[typeEndpoints] = iFactory.Core().V1().Endpoints().Informer()
+	wf.informers[typePolicies] = iFactory.Networking().V1().NetworkPolicies().Informer()
+	wf.informers[typeNamespaces] = iFactory.Core().V1().Namespaces().Informer()
+	wf.informers[typeNodes] = iFactory.Core().V1().Nodes().Informer()
```

Comment:
The work looks good to me. At some point if we were to do selective watching, we will have to introduce something similar to the earlier method, though. Maybe just pass flags on which informers to initialize and which ones to skip. Maybe tie this function to the Run() function at that point.